libs/algebra/include/nil/crypto3/algebra/curves/detail/scalar_mul.hpp

```diff
@@ -32,135 +33,103 @@
 #include <boost/multiprecision/number.hpp>
 #include <nil/crypto3/multiprecision/modular/modular_adaptor.hpp>
 
-#include <cstdint>
+#include <nil/crypto3/multiprecision/cpp_int_modular.hpp>
+
+#include <nil/crypto3/algebra/wnaf.hpp>
 
 namespace nil {
     namespace crypto3 {
         namespace algebra {
             namespace curves {
                 namespace detail {
-                    template<typename GroupValueType,
+                    template<typename curve_element_type,
```

Comment:
Let's rename this to CurveElementType to look similar to the other params.

Reply:
done